init-tools.sh

```diff
@@ -29,53 +16,86 @@ __PACKAGES_DIR=$__scriptpath/packages
-            __DOTNET_PKG="dotnet-dev-$ID.$VERSION_ID-x64"
+# Extended version of platform detection logic from dotnet/cli/scripts/obtain/dotnet-install.sh 16692fc
+get_current_linux_name() {
+    # Detect Distro
+    if [ "$(cat /etc/*-release | grep -cim1 ubuntu)" -eq 1 ]; then
+        if [ "$(cat /etc/*-release | grep -cim1 16.04)" -eq 1 ]; then
+            echo "ubuntu.16.04"
+            return 0
+        fi
+        if [ "$(cat /etc/*-release | grep -cim1 16.10)" -eq 1 ]; then
+            echo "ubuntu.16.10"
+            return 0
         fi
-        ;;
-
-    *)
-        echo "Unsupported OS $OSName detected. Downloading ubuntu-x64 tools"
-        OS=Linux
-        __DOTNET_PKG=dotnet-dev-ubuntu.14.04-x64
-        ;;
-esac
 
-# Initialize Linux Distribution name and .NET CLI package name.
+        echo "ubuntu"
+        return 0
+    elif [ "$(cat /etc/*-release | grep -cim1 centos)" -eq 1 ]; then
+        echo "centos"
+        return 0
+    elif [ "$(cat /etc/*-release | grep -cim1 rhel)" -eq 1 ]; then
+        echo "rhel"
+        return 0
+    elif [ "$(cat /etc/*-release | grep -cim1 debian)" -eq 1 ]; then
+        echo "debian"
+        return 0
+    elif [ "$(cat /etc/*-release | grep -cim1 fedora)" -eq 1 ]; then
+        if [ "$(cat /etc/*-release | grep -cim1 23)" -eq 1 ]; then
+            echo "fedora.23"
+            return 0
+        fi
+        if [ "$(cat /etc/*-release | grep -cim1 24)" -eq 1 ]; then
+            echo "fedora.24"
+            return 0
+        fi
+    elif [ "$(cat /etc/*-release | grep -cim1 opensuse)" -eq 1 ]; then
+        if [ "$(cat /etc/*-release | grep -cim1 13.2)" -eq 1 ]; then
+            echo "opensuse.13.2"
+            return 0
+        fi
+        if [ "$(cat /etc/*-release | grep -cim1 42.1)" -eq 1 ]; then
+            echo "opensuse.42.1"
+            return 0
+        fi
+    fi
 
-initDistroName $OS
+    # Cannot determine Linux distribution, assuming Ubuntu 14.04.
+    echo "ubuntu"
+    return 0
+}
 
-# Work around mac build issue 
-if [ "$OS" == "OSX" ]; then
-  echo Raising file open limit - otherwise Mac build may fail
-  echo ulimit -n 2048
-  ulimit -n 2048
+if [ -z "$__DOTNET_PKG" ]; then
+OSName=$(uname -s)
+    case $OSName in
+        Darwin)
+            OS=OSX
+            __DOTNET_PKG=dotnet-dev-osx-x64
+            ulimit -n 2048
+            ;;
+
+        Linux)
+            __DOTNET_PKG="dotnet-dev-$(get_current_linux_name)-x64"
+            OS=Linux
+            ;;
+
+        *)
+            echo "Unsupported OS '$OSName' detected. Downloading ubuntu-x64 tools."
+            OS=Linux
+            __DOTNET_PKG=dotnet-dev-ubuntu-x64
+            ;;
+  esac
 fi
 
-__CLIDownloadURL=https://dotnetcli.blob.core.windows.net/dotnet/preview/Binaries/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz
+__CLIDownloadURL=https://dotnetcli.blob.core.windows.net/dotnet/Sdk/${__DOTNET_TOOLS_VERSION}/${__DOTNET_PKG}.${__DOTNET_TOOLS_VERSION}.tar.gz
```

Comment:
Interesting about the CDN, if this gets through CI and is ok to merge I'd like to do that in another round of PRs just so these changes can get though.

About removing the duplication, that's discussed here: https://github.com/dotnet/core-setup/pull/254#discussion_r78470982. In short: that'll happen and more, but I don't know where/if it's tracked. /cc @jhendrixMSFT